pkg/api/pluginproxy/ds_proxy.go

```diff
@@ -236,7 +237,12 @@ func (proxy *DataSourceProxy) director(req *http.Request) {
 	req.Header.Del("Referer")
 
 	if proxy.route != nil {
-		ApplyRoute(proxy.ctx.Req.Context(), req, proxy.proxyPath, proxy.route, proxy.ds, proxy.cfg)
+		// Pass in the context with signed-in userId, so that it can be used to obtain user token if the configured token provider supports
+		ctx := proxy.ctx.Req.Context()
+		if proxy.ctx.SignedInUser != nil {
```

Comment:
I think the only use case for having the `Azure` auth type in the ds_auth_provider is that it allows azure auth in data source plugins that don't have a backend. Do you plan to build Azure plugins without a backend? If not, I think we should remove the Azure auth type from the ds_auth_provider. ADX and Azure Monitor don't use routes in plugin.json anymore.

Reply:
Great suggestion! Since the middleware of the httpClient only gets context, the most straightforward option seems to use the context. I will move it from AzureMonitor to the corePlugin.QueryData and corePlugin.CallResource so all corePlugins get this by default in the context. How do you think?

Regarding the proxy path, we do have an internal plugin without backend. Can we keep it for now?

Reply:
Moved the context to corePlugin so individual plugin doesn't need to do this. 
Totally agree that in the future we shall move some common functionalities to a separate SDK as you suggested. Please let me know if this is Ok for now. Thanks!